examples/01-blog/Blog/Type/ImageType.php

```diff
@@ -1,62 +1,63 @@
 <?php
+
+declare(strict_types=1);
+
 namespace GraphQL\Examples\Blog\Type;
 
+use Exception;
 use GraphQL\Examples\Blog\AppContext;
 use GraphQL\Examples\Blog\Data\Image;
 use GraphQL\Examples\Blog\Types;
-use GraphQL\Type\Definition\EnumType;
 use GraphQL\Type\Definition\ObjectType;
+use GraphQL\Type\Definition\ResolveInfo;
+use UnexpectedValueException;
 
 class ImageType extends ObjectType
 {
     public function __construct()
     {
-        $config = [
-            'name' => 'ImageType',
+        parent::__construct([
+            'name' => 'Image',
             'fields' => [
                 'id' => Types::id(),
-                'type' => new EnumType([
-                    'name' => 'ImageTypeEnum',
-                    'values' => [
-                        'USERPIC' => Image::TYPE_USERPIC
-                    ]
-                ]),
-                'size' => Types::imageSizeEnum(),
+                'size' => Types::imageSize(),
                 'width' => Types::int(),
                 'height' => Types::int(),
                 'url' => [
                     'type' => Types::url(),
-                    'resolve' => [$this, 'resolveUrl']
+                    'resolve' => [$this, 'resolveUrl'],
                 ],
 
                 // Just for the sake of example
                 'fieldWithError' => [
                     'type' => Types::string(),
-                    'resolve' => function() {
-                        throw new \Exception("Field with exception");
-                    }
+                    'resolve' => static function (): void {
+                        throw new Exception('Field with exception');
+                    },
                 ],
                 'nonNullFieldWithError' => [
                     'type' => Types::nonNull(Types::string()),
-                    'resolve' => function() {
-                        throw new \Exception("Non-null field with exception");
-                    }
-                ]
-            ]
-        ];
-
-        parent::__construct($config);
+                    'resolve' => static function (): void {
+                        throw new Exception('Non-null field with exception');
+                    },
+                ],
+            ],
+        ]);
     }
 
-    public function resolveUrl(Image $value, $args, AppContext $context)
+    /**
+     * @param array<void> $args
```

Comment:
what does `array<void>` mean?

Reply:
My intention was to specify an empty array type. `array{}` did not work, do you have other suggestions?

Reply:
There are only `non-empty-array`s so no actual way to express empty array AFAIK.

The best way to type $args is IMO `array<string, mixed>`, it is not guaranteed it's empty anyway, is it?

Reply:
In this case it is, since the field has no args. When well-known, I am starting to type `$args` with array shapes such as `array{foo: int, bar: string}`.

Reply:
Yup but the super type is `array<string, mixed>` and it is the closest valid type we have for this no-args thing currently I guess.

Reply:
PHPStan seems to be ok with `array<void>` though, and it does express the intent.

Reply:
PHPStan is still ok with many things like [inserting under string key when int key is declared](https://phpstan.org/r/1f31a564-7fff-475a-9875-c658e1b6078e). Would not rely on it much in cases like this.

I just remember there's `array<empty, empty>` :)

Reply:
PHPStan does not seem to know it yet. I think I opened some related issue there about it but can't find it now.

https://psalm.dev/r/3c66dc8bcf

https://psalm.dev/docs/annotating_code/type_syntax/atomic_types/#empty

Reply:
I went with `array<never>` now, `never` is the bottom type in PHPStan.